HCCGo/app/js/fileManageService.js

```diff
@@ -317,6 +354,22 @@ fileManageService.factory('fileManageService',['$log', '$q', '$routeParams', 'co
    }
    
    service.localHighlight = function(id) {
+      service.localOverwrite = false;
+      service.remoteOverwrite = false;
+      if(id.Class==="ext_txt") {
+        angular.element("#fileStats").show();
+        angular.element("#flocation").text(id.location);
+        angular.element("#fsize").text(id.size);
+        angular.element("#fmtime").text(id.mtime);
+      }
+      else {
+        angular.element("#fileStats").hide();
+      }
+      for(let dirObj of service.remoteFiles) {
+        if(id.name === dirObj.name) {
+          service.remoteOverwrite = true; 
+        }
+      }
```

Comment:
For the sake of optimizations, instead of using this for loop could you look into using the async libraries async.each function to loop through and check everything? There's no dependencies between loops so we can get pseudo-parallel operational speed here with that function. I ask because if someone happens to have a directory of hundreds of files and/or folders I don't want them sitting there waiting while that loop does its thing.